tests/unit/testplan/runnable/interactive/test_api.py

```diff
@@ -255,7 +255,34 @@ def test_put(self, api_env):
         compare_json(json_rsp, json_test, ignored_keys=["runtime_status"])
 
         ihandler.run_test.assert_called_once_with(
-            "MTest1", await_results=False
+            "MTest1", shallow_report=None, await_results=False
+        )
+
+    def test_put_filtered(self, api_env):
+        """Test updating the SingleTest resource via PUT."""
+        client, ihandler = api_env
+
+        json_test = ihandler.report["MTest1"].serialize()
+        json_test["runtime_status"] = report.RuntimeStatus.RUNNING
```

Comment:
The problem I see is the following: the response JSON body is always "shallow" it does not have entries. You cannot do much with it other than top level. We either reduce ourselves to the fact that the filtered branch of exeuction was used (this would be covered by ihandler.run_test.assert_called_once_with as below) or try to find a way to ensure the client here responds with all entries preserved. I do not know how that can be achieved at the moment.